src/main/java/org/apache/ibatis/executor/resultset/DefaultResultSetHandler.java

```diff
@@ -477,7 +477,7 @@ private Object getPropertyMappingValue(ResultSet rs, MetaObject metaResultObject
           }
         }
         final String property = metaObject.findProperty(propertyName, configuration.isMapUnderscoreToCamelCase());
-        if (property != null && metaObject.hasSetter(property)) {
+        if (property != null && metaObject.hasSetter(property) && !resultMap.getMappedProperties().contains(property)) {
```

Comment:
OK..I have to admit that I am a little bit confused 😆
Please let me clarify.

When running the contributed test case (AutomappingTest.shouldGetAUserWhithPhoneNumber()), four columns (id, name, phone and phone_number) are returned from the query and only three of them are mapped.

- 'phone_number' is explicitly mapped
- 'id' and 'name' are auto-mapped

I thought that the unmapped column 'phone' is an _unknown column_, but it actually isn't, is that correct?